cmd/ocitools/generate.go

```diff
@@ -463,3 +413,95 @@ func parseBindMount(s string) (string, string, string, error) {
 
 	return source, dest, options, nil
 }
+
+func addSeccomp(context *cli.Context, g *generate.Generator) error {
+	// Set the DefaultAction of seccomp
+	seccompDefault := context.String("seccomp-default")
+	err := g.SetDefaultSeccompAction(seccompDefault)
+	if err != nil {
+		return err
+	}
+
+	// Add the additional architectures permitted to be used for system calls
+	seccompArch := context.String("seccomp-arch")
+	architectureArgs := strings.Split(seccompArch, ",")
+	for _, arg := range architectureArgs {
+		err = g.SetSeccompArchitecture(arg)
+		if err != nil {
+			return err
+		}
+	}
+
+	if context.IsSet("seccomp-errno") {
+		err := seccompSet(context, "errno", g)
+		if err != nil {
+			return err
+		}
+	}
+
+	if context.IsSet("seccomp-kill") {
+		err := seccompSet(context, "kill", g)
+		if err != nil {
+			return err
+		}
+	}
+
+	if context.IsSet("seccomp-trace") {
+		err := seccompSet(context, "trace", g)
+		if err != nil {
+			return err
+		}
+	}
+
+	if context.IsSet("seccomp-trap") {
+		err := seccompSet(context, "trap", g)
+		if err != nil {
+			return err
+		}
+	}
+
+	if context.IsSet("seccomp-allow") {
+		err := seccompSet(context, "allow", g)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func seccompSet(context *cli.Context, seccompFlag string, g *generate.Generator) error {
+	flagInput := context.String("seccomp-" + seccompFlag)
+	flagArgs := strings.Split(flagInput, ",")
+	setSyscallArgsSlice := []seccomp.SyscallOpts{}
+	for _, flagArg := range flagArgs {
+		comparisonArgs := strings.Split(flagArg, ":")
+		if len(comparisonArgs) == 5 {
+			setSyscallArgs := seccomp.SyscallOpts{
+				Action:   seccompFlag,
+				Syscall:  comparisonArgs[0],
+				Index:    comparisonArgs[1],
+				Value:    comparisonArgs[2],
+				ValueTwo: comparisonArgs[3],
+				Operator: comparisonArgs[4],
+			}
+			setSyscallArgsSlice = append(setSyscallArgsSlice, setSyscallArgs)
+		} else if len(comparisonArgs) == 1 {
+			setSyscallArgs := seccomp.SyscallOpts{
+				Action:  seccompFlag,
+				Syscall: comparisonArgs[0],
+			}
+			setSyscallArgsSlice = append(setSyscallArgsSlice, setSyscallArgs)
+		} else {
+			return errors.New("Invalid syscall argument formatting")
```

Comment:
@GrantSeltzer , you may want to try `fmt.Errorf` here. It allows you to output which `flagArg` is broken.
```
func Errorf(format string, a ...interface{}) error
```